src/testing/TGV_scaling.cpp

```diff
@@ -0,0 +1,180 @@
+    std::array<double,poly_degree_end> time_to_run_mpi;
+
+    //poly degree loop
+    for(unsigned int poly_degree = poly_degree_start; poly_degree<poly_degree_end; poly_degree++){
+
+        // set the warped grid
+        const unsigned int grid_degree = (all_parameters->use_curvilinear_grid) ? poly_degree : 1;
+         
+        if(all_parameters->overintegration == 100){
+            if(all_parameters->use_curvilinear_grid){
+                all_parameters_new.overintegration = 2*(poly_degree+1);
+            }
+            else{
+                all_parameters_new.overintegration = poly_degree+1;
+            }
+        }
+         
+        // Create DG
+        std::shared_ptr < PHiLiP::DGBase<dim, double> > dg = PHiLiP::DGFactory<dim,double>::create_discontinuous_galerkin(&all_parameters_new, poly_degree, poly_degree, grid_degree, grid);
+        dg->allocate_system (false,false,false);
+         
+        //Apply initial condition for TGV
+        std::shared_ptr< InitialConditionFunction<dim,nstate,double> > initial_condition_function = 
+                    InitialConditionFactory<dim,nstate,double>::create_InitialConditionFunction(&all_parameters_new);
+        SetInitialCondition<dim,nstate,double>::set_initial_condition(initial_condition_function, dg, &all_parameters_new);
+        //Create ODE system. 
+        std::shared_ptr<ODE::ODESolverBase<dim, double>> ode_solver = ODE::ODESolverFactory<dim, double>::create_ODESolver(dg);
+         
+        ode_solver->current_iteration = 0;
+        ode_solver->allocate_ode_system();
+        MPI_Barrier(MPI_COMM_WORLD);
+        pcout << "ODE solver successfully created. This verifies no memory jump from ODE Solver." << std::endl;
+        dealii::LinearAlgebra::distributed::Vector<double> solution_update;
+        solution_update.reinit(dg->locally_owned_dofs, dg->ghost_dofs, this->mpi_communicator);
+
+        //Perform 10 steps solving the reidual and applying the inverse of the mass matrix.
+        for(unsigned int i_step=0; i_step<10; i_step++){
+            dg->assemble_residual();
+            if(all_parameters->use_inverse_mass_on_the_fly){
+                dg->apply_inverse_global_mass_matrix(dg->right_hand_side, solution_update);
+            } else{
+                dg->global_inverse_mass_matrix.vmult(solution_update, dg->right_hand_side);
+            }
+        }
+
+        //store local cpu time
+        time_to_run[poly_degree] = dg->assemble_residual_time;
+        //store mpi summed cpu time
+        time_to_run_mpi[poly_degree] = dealii::Utilities::MPI::sum(time_to_run[poly_degree], this->mpi_communicator);
+        
+        std::cout<<"Poly Degree "<<poly_degree<<" time to run local cpu "<<std::fixed << std::setprecision(16) << (double)time_to_run[poly_degree]<<std::endl;
+        pcout<<"Poly Degree "<<poly_degree<<" time to run Mpi "<<std::fixed << std::setprecision(16) << (double)time_to_run_mpi[poly_degree]<<std::endl;
```

Comment:
Maybe add an MPI barrier here - otherwise the MPI output line can print in the middle of the "local cpu" times.

Reply:
Yes added the mpi barrier.